pkgs/applications/networking/cluster/cmctl/default.nix

```diff
@@ -2,33 +2,49 @@
 
 buildGoModule rec {
   pname = "cmctl";
-  version = "1.8.2";
+  version = "1.9.1";
 
   src = fetchFromGitHub {
     owner = "cert-manager";
-    repo = "cert-manager";
-    rev = "v${version}";
-    sha256 = "sha256-sfC1acnCrcQ4A1tXXcjh47Af6xeJqjdGXy0gK21ZSFg=";
+    repo  = "cert-manager";
+    rev   = "v${version}";
+    hash  = "sha256-Z1aJ18X4mfJPlCPBC7QgfdX5Tk4+PK8mYoJZhGwz9ec=";
   };
 
-  vendorSha256 = "sha256-UYw9WdQ6VwzuuiOsa1yovkLZG7NmLYSW51p8UhmQMeI=";
+  vendorSha256 = "sha256-45+tZZAEHaLdTN1NQCueJVTx5x2IanwDl+Y9MELqdBE=";
 
   subPackages = [ "cmd/ctl" ];
 
-  ldflags = [ "-s" "-w" ];
+  ldflags = [
+    "-s" "-w"
+    # See https://github.com/cert-manager/cert-manager/blob/4486c01f726f17d2790a8a563ae6bc6e98465505/make/cmctl.mk#L1
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build.name=cmctl"
+    "-X" "github.com/cert-manager/cert-manager/cmd/ctl/pkg/build/commands.registerCompletion=true"
+  ];
 
   nativeBuildInputs = [ installShellFiles ];
 
+  # Ensure we capture runtime completions by calling `completion` subcommand.
   postInstall = ''
     mv $out/bin/ctl $out/bin/cmctl
-    installShellCompletion --cmd cmctl \
-      --bash <($out/bin/cmctl completion bash) \
-      --fish <($out/bin/cmctl completion fish) \
-      --zsh <($out/bin/cmctl completion zsh)
+    $out/bin/cmctl completion bash > cmctl.bash
+    $out/bin/cmctl completion zsh  > cmctl.zsh
+    $out/bin/cmctl completion fish > cmctl.fish
+    installShellCompletion cmctl.{bash,zsh,fish}
```

Comment:
My expectation is the previous code to be equivalent to the code you are suggestion. I couldn't reproduce problem. So I'd like to confirm that this is really causing a problem before we change it.

Reply:
So by "runtime completions" I mean CLI completions whereby suggestions are derived at runtime, rather than just the static sub-command suggestions. In `cmctl`s case, this is Kubernetes resources.

As an example, if I type the following command and press  `<TAB>`, I get suggestions for the current list of Namespaces:

```terminal
$ cmctl approve -n
another-namespace   cert-manager        default             kube-node-lease     kube-public         kube-system         local-path-storage  my-namespace
```

This functionality was missing before this PR, and pressing `<TAB>` wouldn't show anything.

Reply:
Hmm, taking another look I understand your point now. I'm doing the same thing. 

I'll double check that it definitely wasn't working before again. I agree, your previous method is more concise.

Reply:
Yes- sorry for the noise here. I must have gotten myself confused with another package..

I've reverted back the changes.

Reply:
But what is causing this? Is really the way `installShellCompletion` is being executed?

Reply:
No, everything was working correctly the way it was before. For whatever reason, I got it into my head that it needed changing and didn't read the previous command properly.